contracts/log/log.go

```diff
@@ -41,6 +43,16 @@ type Writer interface {
 	Fatalf(format string, args ...any)
 	Panic(args ...any)
 	Panicf(format string, args ...any)
+	User(userID string, userData ...map[string]any) Writer
+	Owner(ownerID string) Writer
+	Hint(hint string) Writer
+	Trace(trace string) Writer
```

Comment:
Maybe we can remove this method, adding the trace information when `Error` automatically.